src/renderers/shared/fiber/ReactFiberScheduler.js

```diff
@@ -345,7 +348,7 @@ module.exports = function<T, P, I, TI, C, CX>(config : HostConfig<T, P, I, TI, C
       firstEffect = finishedWork.firstEffect;
     }
 
-    prepareForCommit();
+    const commitInfo = prepareForCommit();
```

Comment:
I reverted your most recent commit (79f01b2) and ran all tests, and you're right. Looking at one of the failing tests, it seems that unmounting from a `componentWillUnmount` hook eventually results in more sync priority work being scheduled.

Okay. You're right, this seems less fragile anyway. 👍